src/export/exporter.ts

```diff
@@ -2,38 +2,52 @@ import { ExportResult } from '@opentelemetry/base';
 import { Logger } from '@opentelemetry/api';
 import { NoopLogger } from '@opentelemetry/core';
 import { Envelope } from '../Declarations/Contracts';
-import { NoopSender } from '../platform';
-import { ExporterConfig, DEFAULT_EXPORTER_CONFIG } from '../config';
+import { HttpSender } from '../platform';
+import {
+  DEFAULT_EXPORTER_CONFIG,
+  PartialAzureExporterConfigWithSetupString,
+  AzureExporterConfigWithSetupString,
+} from '../config';
 import { BaseExporter, TelemetryProcessor } from '../types';
-import { ArrayPersist } from '../platform/nodejs/arrayPersist';
-import { isRetriable } from '../utils/breezeUtils';
+import { ArrayPersist } from '../platform/nodejs/persist/arrayPersist';
+import { isRetriable, BreezeResponse } from '../utils/breezeUtils';
 
 export abstract class AzureMonitorBaseExporter implements BaseExporter {
-  private readonly _persister: ArrayPersist<Envelope[]>; // @todo: replace with FileSystemPersister
+  protected readonly _persister: ArrayPersist<Envelope[]>; // @todo: replace with FileSystemPersister
 
   protected readonly _logger: Logger;
 
-  private readonly _sender: NoopSender;
+  protected readonly _sender: HttpSender;
+
+  protected _retryTimer: NodeJS.Timeout | null;
 
   protected _telemetryProcessors: TelemetryProcessor[];
 
-  constructor(options: ExporterConfig = DEFAULT_EXPORTER_CONFIG) {
-    this._logger = options.logger || new NoopLogger();
+  private readonly _options: AzureExporterConfigWithSetupString;
+
+  constructor(_options: PartialAzureExporterConfigWithSetupString = DEFAULT_EXPORTER_CONFIG) {
+    this._logger = _options.logger || new NoopLogger();
+    this._options = {
+      ...DEFAULT_EXPORTER_CONFIG,
+      instrumentationKey: undefined, // hack: clear empty string ikey from default config
```

Comment:
Why not having undefined as default value instead?

Reply:
Removed this line because config interface is simpler now